src/main/daml/Daml/Finance/Interface/Lifecycle/Rule/NewVersion.daml

```diff
@@ -0,0 +1,55 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Lifecycle.Rule.NewVersion where
+
+-- | Type synonym for `NewVersion`.
+type I = NewVersion
+
+-- | Type synonym for `View`.
+type V = View
+
+-- | View for `NewVersion`.
+data View = View
+  with
+    lifecycler : Party
+      -- ^ Party performing the lifecycling.
+  deriving (Eq, Show)
+
+-- | Interface implemented by instruments that create a contingent claims tree on-the-fly.
+interface NewVersion where
```

Comment:
This feels more like an instrument interface and I would consider `Interface.Claims` to be a better place for it.

Also, I am trying to think about a more descriptive name than `NewVersion`: the name of the interface doesn't really hint at what is in the description

Reply:
> This feels more like an instrument interface and I would consider `Interface.Claims` to be a better place for it.

I have moved it now.

Reply:
> Also, I am trying to think about a more descriptive name than `NewVersion`: the name of the interface doesn't really hint at what is in the description

The interface is all about evolving the instrument by moving `lastEventTimestamp` forward. Maybe something like `FastForward`?

Reply:
Mmm, at some point the interface will need to include additional information, such as the history of relevant lifecycle times + elections.

Reply:
I wonder if we should call it just `Instrument` and put it in a sub-folder that highlights the fact that claims are generated on-the-fly

Reply:
Something like `Daml.Finance.Interface.Claims.OnTheFly.Instrument ` ?

Reply:
I think that could work, as long as we are descriptive in the interface descriptions. @georg-da any strong opinion here?

Reply:
We can also discuss it in today's catchup.

Reply:
I'd maybe vote for `Daml.Finance.Interface.Claims.Dynamic.Instrument`